ignite/handlers/param_scheduler.py

```diff
@@ -1008,16 +1017,17 @@ def print_lr():
             param_group_warmup_end_value = warmup_end_value
 
         milestones_values = [(0, warmup_start_value), (warmup_duration - 1, param_group_warmup_end_value)]
-
         if isinstance(lr_scheduler, _LRScheduler):
             init_lr = param_group["lr"]
 
             if init_lr != param_group_warmup_end_value:
                 milestones_values.append((warmup_duration, init_lr))
 
+            lr_scheduler.last_epoch = 1
```

Comment:
I set `lr_scheduelr.last_epoch` to one here so that the initial `lr` value `of optimizer` avoids to overlap with `milestones_values`.

Reply:
@yuta0821 I'd avoid hard-coding values like that, can we make it relative somehow ? Why it should be explicitly as 1 and not 2 or 3 ?

Reply:
@yuta0821 how about here, if we were doing `lr_scheduler.last_epoch += 1`, instead of `lr_scheduler.last_epoch = 1` ?

Reply:
> @yuta0821 I'd avoid hard-coding values like that, can we make it relative somehow ? Why it should be explicitly as 1 and not 2 or 3 ?

I chose 1 to skip the extra initial value which is equal the value of `last_epoch = 0`.

> yuta0821 how about here, if we were doing lr_scheduler.last_epoch += 1, instead of lr_scheduler.last_epoch = 1 ?

When the `create_lr_scheduler_with_warmup` is constructed, `lr_scheduler.last_epoch = 0` is assumed, so `lr_scheduler.last_epoch = 1` is same as `lr_scheduler.last_epoch += 1`.

Reply:
`lr_scheduler` is an input argument where user can manually set `lr_scheduler.last_epoch` to something different to 0. If we set it internally to 1 this can be a "surprising" behaviour. I do not know why user would want to set it to something (maybe while restoring from a state), but setting it to 1 can't be that innocent, I think.

Reply:
Adding `lr_scheduler.last_epoch += 1`  to skip the initial `lr` value can't be that innocent, either ?
I think this part of code plays the same role as following code in `LRScheduler.__init__`.
```python
if use_legacy:
    warnings.warn(
        "Please make sure to attach scheduler to Events.ITERATION_COMPLETED"
        "instead of Events.ITERATION_STARTED to make sure to use"
        "the first lr value from the optimizer, otherwise it is will be skipped"
     )
    self.lr_scheduler.last_epoch += 1  # type: ignore[attr-defined]
```

Reply:
That's true, if we do `lr_scheduler.last_epoch += 1` or `lr_scheduler.last_epoch = 1` it would be equivalent to `use_legacy=True` once `lr_scheduler` is wrapped with `LRScheduler`.